py3status/modules/gcal.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+"""
+Display upcoming calendar events from Google Calendar.
+
+Configuration parameters:
+    agenda_days: show calendar events for number of days (default 1)
+    button_refresh: mouse button to refresh this module (default 2)
+    cache_timeout: refresh interval for this module (default 21600)
+    format: display format for this module (default '{format_event}')
+    format_event: display format for events
+        (default '[\?if=is_date&color=degraded %a %b %d ]
+        [\?if=is_time&color=good %-I:%M %p ][\?color=bad {event}]')
+    format_separator: show separator only if more than one (default ', ')
+    remove: list of strings to be removed from the output (default [])
+
+Format placeholders:
+    {format_event} format for events
+    is_event: a boolean based on available events
+
+format_event placeholders:
+    {event} scheduled event eg Pizza night
+    %?  strftime characters to be translated. See `man strftime'.
+    is_time: a boolean based on event containing a time schedule
+    is_date: a boolean based on event containing a date schedule
+
+Requires:
+    gcalcli: Google Calendar Command Line Interface
+
+Examples:
+```
+# change event colors
+gcal {
+    color_bad = '#ad7fa8'
+    color_degraded = '#fce94f'
+    color_good = '#34e2e2'
+}
+# show "No Events"
+gcal {
+    format = u'[\?if=is_event {format_event}|\?color=bad No Events]'
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+[
+    {'color': '#FFFF00', 'full_text': u'Fri Jul 20 '},
+    {'color': '#FF0000', 'full_text': u'Tacos'},
+    {'full_text': u', '},
+    {'color': '#FF0000', 'full_text': u'Volleyball'},
+    {'full_text': u', '},
+    {'color': '#00FF00', 'full_text': u'6:00 AM '},
+    {'color': '#FF0000', 'full_text': u'Garbage Pickup'},
+    {'full_text': u', '}
+    {'color': '#FFFF00', 'full_text': u'Sat Jul 21 '},
+    {'color': '#00FF00', 'full_text': u'6:05 PM '},
+    {'color': '#FF0000', 'full_text': u'Flight to Poland'}
+]
+"""
+
+from datetime import datetime as dt, timedelta
+import re
+
+STRING_NOT_INSTALLED = "isn't installed"
+FMT_DATE = '%a %b %d'
+FMT_GCAL = '%Y%m%d'
+FMT_SS = '%s %s'
+FMT_TIME = '%I:%M%p'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    agenda_days = 1
+    button_refresh = 2
+    cache_timeout = 21600
```

Comment:
> People might like having events there from few hours ago. 

there are better ways to achieve this if it is wanted.  With as is now the time depends on when the refresh happens which is super inconsistent.